### ISSUE TITLE
DDLDatabase defaultNameCase property shouldn't affect built in functions in view definitions

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12752]
Using the `DDLDatabase` to parse a view like this:

```sql
CREATE VIEW v AS
SELECT COALESCE(SUM(t.a), 0) AS sum
FROM (VALUES (1)) AS t(a);
```

And then configuring the code generator like this, to produce lower case identifiers, e.g. for PostgreSQL usage:

```xml
<configuration>
    <generator>
        <database>
            <name>org.jooq.meta.extensions.ddl.DDLDatabase</name>
            <properties>
                <property>
                    <key>scripts</key>
                    <value>${basedir}/src/main/resources/viewswithnamecase.sql</value>
                </property>
                <property>
                    <key>defaultNameCase</key>
                    <value>lower</value>
                </property>
            </properties>
        </database>
        <target>
            <packageName>org.jooq.codegen.test.viewswithlongsources.db</packageName>
        </target>
    </generator>
</configuration>
```

Produces an exception, because of how we implemented the `DDLDatabase`:

```
Caused by: org.h2.jdbc.JdbcSQLSyntaxErrorException: Funktion "coalesce" nicht gefunden
Function "coalesce" not found; SQL statement:
create view "v" as select "coalesce"("sum"("t"."a"), 0) "sum" from (values (1)) "t" ("a") [90022-202]
    at org.h2.message.DbException.getJdbcSQLException (DbException.java:632)
    at org.h2.message.DbException.getJdbcSQLException (DbException.java:477)
    at org.h2.message.DbException.get (DbException.java:223)
    at org.h2.message.DbException.get (DbException.java:199)
    at org.h2.command.Parser.readFunction (Parser.java:4179)
    at org.h2.command.Parser.readTerm (Parser.java:5521)
    at org.h2.command.Parser.readFactor (Parser.java:3726)
    at org.h2.command.Parser.readSum (Parser.java:3713)
    at org.h2.command.Parser.readConcat (Parser.java:3678)
    at org.h2.command.Parser.readCondition (Parser.java:3461)
    at org.h2.command.Parser.readExpression (Parser.java:3381)
    at org.h2.command.Parser.parseSelectExpressions (Parser.java:3196)
    at org.h2.command.Parser.parseSelect (Parser.java:3213)
    at org.h2.command.Parser.parseQuerySub (Parser.java:3092)
    at org.h2.command.Parser.parseSelectUnion (Parser.java:2930)
    at org.h2.command.Parser.parseQuery (Parser.java:2900)
    at org.h2.command.Parser.parseCreateView (Parser.java:8565)
    at org.h2.command.Parser.parseCreate (Parser.java:7750)
    at org.h2.command.Parser.parsePrepared (Parser.java:1140)
    at org.h2.command.Parser.parse (Parser.java:1068)
    at org.h2.command.Parser.parse (Parser.java:1037)
    at org.h2.command.Parser.prepareCommand (Parser.java:965)
    at org.h2.engine.SessionLocal.prepareLocal (SessionLocal.java:614)
    at org.h2.engine.SessionLocal.prepareCommand (SessionLocal.java:552)
    at org.h2.jdbc.JdbcConnection.prepareCommand (JdbcConnection.java:1111)
    at org.h2.jdbc.JdbcStatement.executeInternal (JdbcStatement.java:237)
    at org.h2.jdbc.JdbcStatement.execute (JdbcStatement.java:223)
    at org.jooq.tools.jdbc.DefaultStatement.execute (DefaultStatement.java:102)
    at org.jooq.impl.SettingsEnabledPreparedStatement.execute (SettingsEnabledPreparedStatement.java:227)
    at org.jooq.impl.AbstractQuery.execute (AbstractQuery.java:414)
    at org.jooq.impl.AbstractQuery.execute (AbstractQuery.java:335)
    at org.jooq.meta.extensions.ddl.DDLDatabase.load (DDLDatabase.java:183)
    at org.jooq.meta.extensions.ddl.DDLDatabase.lambda$export$0 (DDLDatabase.java:156)
    at org.jooq.FilePattern.load0 (FilePattern.java:307)
    at org.jooq.FilePattern.load (FilePattern.java:287)
    at org.jooq.FilePattern.load (FilePattern.java:234)
    at org.jooq.meta.extensions.ddl.DDLDatabase.export (DDLDatabase.java:156)
    at org.jooq.meta.extensions.AbstractInterpretingDatabase.connection (AbstractInterpretingDatabase.java:100)
    at org.jooq.meta.extensions.AbstractInterpretingDatabase.create0 (AbstractInterpretingDatabase.java:77)
    at org.jooq.meta.AbstractDatabase.create (AbstractDatabase.java:337)
    at org.jooq.meta.AbstractDatabase.create (AbstractDatabase.java:327)
    at org.jooq.meta.AbstractDatabase.setConnection (AbstractDatabase.java:317)
    at org.jooq.codegen.GenerationTool.run0 (GenerationTool.java:535)
    at org.jooq.codegen.GenerationTool.run (GenerationTool.java:238)
    at org.jooq.codegen.GenerationTool.generate (GenerationTool.java:233)
    at org.jooq.codegen.maven.Plugin.execute (Plugin.java:207)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:77)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:568)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

The setting transforms *all* identifiers to quoted, lower case identifiers, even the ones referencing built-in functions, which is obviously incorrect. See a similar issue here: https://github.com/jOOQ/jOOQ/issues/9931, which isn't the same, because the `DDLDatabase` uses a `VisitListener` to transform all identifiers.

I'm not sure if there's a good way to fix this particular flag. It's kind of a hack to quote all identifiers contained in user SQL this way. If we interpreted the SQL ourselves, instead of running it through H2, we could probably emulate the desired target behaviour much better.

**Comment by lukaseder** [comment]
On the other hand, if we did introduce those system names as suggested in #9931, then we could avoid the change of case also for system names in this case here.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0. Since this depends on new API and a change of behaviour of existing `unquotedName()` vs `systemName()` behaviour in a variety of functionality, I won't backport this to 3.15.
